subprojects/docs/src/docs/userguide/gradleWrapper.adoc

```diff
@@ -0,0 +1,184 @@
+* `gradlew &lt;task&gt;` (on Windows using the gradlew.bat batch file)
+
+Each Wrapper is tied to a specific version of Gradle, so when you first run one of the commands above for a given Gradle version, it will download the corresponding Gradle distribution and use it to execute the build.
+
+[TIP]
+.IDEs
+====
+
+When importing a Gradle project via its wrapper, your IDE may ask to use the Gradle 'all' distribution. This is perfectly fine and helps the IDE provide code completion for the build files.
+
+====
+
+Not only does this mean that you don’t have to manually install Gradle yourself, but you are also sure to use the version of Gradle that the build is designed for. This makes your historical builds more reliable. Just use the appropriate syntax from above whenever you see a command line starting with `gradle ...` in the user guide, on Stack Overflow, in articles or wherever.
+
+For completeness sake, and to ensure you don’t delete any important files, here are the files and directories in a Gradle project that make up the Wrapper:
+
+* `gradlew` (Unix Shell script)
+* `gradlew.bat` (Windows batch file)
+* `gradle/wrapper/gradle-wrapper.jar` (Wrapper JAR)
+* `gradle/wrapper/gradle-wrapper.properties` (Wrapper properties)
+
+
+If you’re wondering where the Gradle distributions are stored, you’ll find them in your user home directory under `$USER_HOME/.gradle/wrapper/dists`.
+
+[[sec:wrapper_generation]]
+=== Adding the Wrapper to a project
+
+The Wrapper is something you _should_ check into version control. By distributing the Wrapper with your project, anyone can work with it without needing to install Gradle beforehand. Even better, users of the build are guaranteed to use the version of Gradle that the build was designed to work with. Of course, this is also great for http://en.wikipedia.org/wiki/Continuous_integration[continuous integration] servers (i.e. servers that regularly build your project) as it requires no configuration on the server.
+
+You install the Wrapper into your project by running the `wrapper` task. (This task is always available, even if you don't add it to your build). To specify a Gradle version use `--gradle-version` on the command-line. By default, the Wrapper will use a `bin` distribution. This is the smallest Gradle distribution. Some tools, like Android Studio and Intellij IDEA, provide additional context information when used with the `all` distribution. You may select a different Gradle distribution type by using `--distribution-type`. You can also set the URL to download Gradle from directly via `--gradle-distribution-url`. If no version or distribution URL is specified, the Wrapper will be configured to use the gradle version the `wrapper` task is executed with. So if you run the `wrapper` task with Gradle 2.4, then the Wrapper configuration will default to version 2.4.
+
+++++
+<sample id="wrapperCommandLine" dir="userguide/wrapper/simple" title="Running the Wrapper task">
```

Comment:
Can you properly indent the sample tags? It will make it more readable for future maintainers.